src/services/MeasureService.ts

```diff
@@ -88,4 +89,35 @@ export class MeasureService implements Service<fhir4.Measure> {
 
     return createMeasurePackageBundle(measure[0]);
   }
+
+  /**
+   * result of sending a POST or GET request to:
+   * {BASE_URL}/4_0_1/Measure/$data-requirements or {BASE_URL}/4_0_1/Measure/:id/$data-requirements
+   * creates a Library with all data requirements for the specified measure
+   * requires parameters id and/or url and/or identifier, but also supports version as supplemental (optional)
+   */
+  async dataRequirements(args: RequestArgs, { req }: RequestCtx) {
+    logger.info(`${req.method} ${req.path}`);
+    logger.info('Using package to create measure bundle');
+    const path = req.path.replace(/data-requirements/i, 'package');
+    const measureBundle = await this.package(args, { req: { ...req, path: path } });
+
+    // TODO: Clarify should period start/end be required or have intelligent defaults in fqm-execution to measure effectivePeriod? (fqm-execution defaults to 2019)
+    // short term: default to 2022 if these aren't passed through
+    const params = gatherParams(req.query, args.resource);
+    const start = params.periodStart || '2022-01-01';
+    const end = params.periodEnd || '2022-12-31';
+
+    logger.info(`Calculating with start ${start} and end ${end}`);
+    const { results } = await Calculator.calculateDataRequirements(measureBundle, {
+      measurementPeriodStart: start,
+      measurementPeriodEnd: end
+    });
+    logger.info('Successfully generated $data-requirements report');
+    return results;
+
+    // TODO: Clarify return type from documentation (https://hl7.org/fhir/us/cqfmeasures/STU3/OperationDefinition-Measure-data-requirements.html)
+    // Documentation describes as a library, but the specified type is "Bundle".
+    // For now, this code returns a Library (similar to the Measure operation)
```

Comment:
I just realized that Matt said in Slack that the return type was confirmed to be Library (see Jira ticket [here](https://jira.hl7.org/browse/FHIR-40226) with the correction). Do we think it's safe to remove this TODO?

Reply:
I agree, Sarah. I think we can remove that TODO.